packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -225,6 +225,45 @@ export interface CanaryProps {
    */
   readonly test: Test;
 
+  /**
+   * Key-value pairs that the Synthetics caches and makes available for your canary
+   * scripts. Use environment variables to apply configuration changes, such
+   * as test and production environment configurations, without changing your
+   * Canary script source code.
+   *
+   * @default - No environment variables.
+   */
+  readonly environment?: { [key: string]: string };
+
+  /**
+   * The function execution time (in seconds) after which Lambda terminates
+   * the function. Because the execution time affects cost, set this value
+   * based on the function's expected execution time.
+   *
+   * @default cdk.Duration.seconds(900)
+   */
+  readonly timeout?: cdk.Duration;
```

Comment:
There are important caveats mentioned in the cloudformation documentation 
`You can't set this time to be longer than the frequency of the runs of this canary.`
`If you omit this field, the frequency of the canary is used as this value, up to a maximum of 900 seconds. `
See: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-runconfig.html#cfn-synthetics-canary-runconfig-timeoutinseconds

Please reflect this in your documentation

Reply:
It should be in the documentation as well as in the code, if the provided value is not valid we should throw an error. There has been a lot of discussion about this feature in https://github.com/aws/aws-cdk/issues/9300

Reply:
Added to doc and commented in the code